src/System.Private.ServiceModel/src/System/ServiceModel/Dispatcher/StreamFormatter.cs

```diff
@@ -94,7 +103,10 @@ internal class SerializeAsyncResult : AsyncResult
             {
                 _streamFormatter = streamFormatter;
                 _writer = writer;
-                throw ExceptionHelper.PlatformNotSupported("StreamFormatter.SerializeAsyncResult is not supported.");
+
+                // As we use the Task-returning method for async operation,
+                // we shouldn't get to this point. Throw exception just in case.
+                throw new NotImplementedException();
```

Comment:
I will throw the exception in the standard way. I thought it would be better to make the method throw NotImplemented.ByDesign for now just in case on any path we call this method.